sdk/eventhub/Azure.Messaging.EventHubs/src/Compatibility/TrackOneEventHubProducer.cs

```diff
@@ -59,17 +59,24 @@ public TrackOneEventHubProducer(Func<TrackOne.EventDataSender> trackOneSenderFac
         /// <param name="sendOptions">The set of options to consider when sending this batch.</param>
         /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
         ///
-        public override Task SendAsync(IEnumerable<EventData> events,
-                                       SendOptions sendOptions,
-                                       CancellationToken cancellationToken)
+        public override async Task SendAsync(IEnumerable<EventData> events,
+                                             SendOptions sendOptions,
+                                             CancellationToken cancellationToken)
         {
             static TrackOne.EventData TransformEvent(EventData eventData) =>
                 new TrackOne.EventData(eventData.Body.ToArray())
                 {
                     Properties = eventData.Properties
                 };
 
-            return TrackOneSender.SendAsync(events.Select(TransformEvent), sendOptions?.PartitionKey);
+            try
+            {
+                await TrackOneSender.SendAsync(events.Select(TransformEvent), sendOptions?.PartitionKey);
```

Comment:
Is this returned by default without the need to explicitly use a return statement?

Reply:
We need to `await` in order for any exception to be unwrapped in this context, so that we can do the mapping to the new track two type.   Otherwise, yes, we could (and were) returning the task directly.

Reply:
That said, I just realized that I missed the `ConfigureAwait(false)` here now that we're awaiting.   Will add that in the next revision.